### PR TITLE
feat: Shows an alert while replying if chats are not assigned to them

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

Creates a new banner component and adds the ability to show an alert to the agents in all conversations if those are not assigned to them and they are trying to reply to it.

Fixes https://github.com/chatwoot/chatwoot/issues/2303

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**Screen recording**
 
https://user-images.githubusercontent.com/64252451/151491295-4063ad95-c6fe-4e7b-bdf3-b2297c55259a.mov

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by iamsivin** [comment]
@nithindavid Can you take a look?

**Review by fayazara — APPROVED** [review]
Looks good!

File: app/javascript/dashboard/components/ui/Banner.vue [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    },
+    onClickClose(e) {
+      this.$emit('close', e);
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+.banner {
```
Comment (nithindavid) [inline app/javascript/dashboard/components/ui/Banner.vue:82]: Check if there are other occurrences of .banner styles after moving. If it's a big effort we will track in a separate issue.

**Review by nithindavid — APPROVED** [review]
Good work 👍

**Comment by pranavrajs** [comment]
@iamsivin If it is a private note, you don't need to show the message.

File: app/javascript/dashboard/components/ui/Banner.vue [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    },
+    hasActionButton: {
+      type: Boolean,
+      default: false,
+    },
+    actionButtonLabel: {
+      type: String,
+      default: '',
+    },
+    bgColorScheme: {
```
Comment (nithindavid) [inline app/javascript/dashboard/components/ui/Banner.vue:61]:
```suggestion
    ColorScheme: {
```

Color-scheme is enough.

**Review by nithindavid — CHANGES_REQUESTED** [review]

File: app/javascript/dashboard/components/ui/Banner.vue (hunk 2 above)
Comment (iamsivin) [reply app/javascript/dashboard/components/ui/Banner.vue]: Fixes by https://github.com/chatwoot/chatwoot/pull/3867/commits/dbbd0bc22944ac95dd4e78e79d8cc5893f42e303

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**